server/app/api.py

```diff
@@ -1191,24 +1192,100 @@ class SearchAPI(APIResource):
             }
```

Comment:
I think there should be a default `num_per_page` and a default `page`. I think the default `page` should be the first page and the default `num_per_page` can either be a small number so the query goes fast (e.g. 10) or a really large number so you get _everything_ for most queries (e.g. 10000).